### PR TITLE
Create UI component to replace existing video options buttons

### PR DESCRIPTION
**Pull request by robwayne**
### Checklist
- [x] I've run `bundle exec fastlane test` from the root directory to see all new and existing tests pass
- [x] I've followed the [vlc-ios code style](Docs/CodingStyle.md)
- [x] I've read the [Contribution Guidelines](https://github.com/videolan/vlc-ios#contribute)
- [x] I've updated the documentation if necessary.

### CODE REVIEW

File: Sources/VLCMovieViewController.m [hunk 1]
```diff
@@ -468,31 +482,45 @@ - (void)viewDidLayoutSubviews
...
+    #else
+        CGRect videoOptionsBarFrame = CGRectMake(0., 0., 24., 120.);
+        CGFloat x,y;
+        x = controllerPanelFrame.size.width - videoOptionsBarFrame.size.width;
+        y = controllerPanelFrame.origin.y - videoOptionsBarFrame.size.height;
+        videoOptionsBarFrame = CGRectMake(x,y,videoOptionsBarFrame.size.width,videoOptionsBarFrame.size.height);
+        _videoOptionsControlBar.frame = videoOptionsBarFrame;
+        if(UIInterfaceOrientationIsLandscape([[UIApplication sharedApplication] statusBarOrientation])) {
+            _videoOptionsControlBar.orientationAxis = UILayoutConstraintAxisHorizontal;
+        }
```
Comment (Mikanbu) [inline Sources/VLCMovieViewController.m]: Cosmetic nit-pick: Spaces missing after commas and keyword.

File: Sources/VLCMovieViewController.m [hunk 2]
```diff
@@ -644,26 +672,49 @@ - (void)sleepTimerInitializer
...
+                 _controllerPanel.playPauseButton,
+                 _controllerPanel.fwdButton,
+                 _controllerPanel.videoFilterButton,
+                 _videoOptionsControlBar.toggleFullScreenButton,
+                 _videoOptionsControlBar.selectSubtitleButton,
+                 _videoOptionsControlBar.moreOptionsButton,
+                 _videoOptionsControlBar.repeatButton,
+                 _controllerPanel.volumeView,
+                 _rendererButton];
+    #endif
```
Comment (Mikanbu) [inline Sources/VLCMovieViewController.m]: I believe this part could be simplified in order to avoid adding duplication.

File: Sources/VLCMovieViewController.m [hunk 3]
```diff
@@ -754,7 +817,13 @@ - (void)setControlsHidden:(BOOL)hidden animated:(BOOL)animated
         self->_equalizerView.hidden = YES;
         if (self->_sleepTimerContainer)
             self->_sleepTimerContainer.hidden = YES;
-        self->_multiSelectionView.hidden = YES;
+        
+        #if !NEW_UI
+                self->_multiSelectionView.hidden = YES;
```
Comment (Mikanbu) [inline Sources/VLCMovieViewController.m]: Cosmetic nit-pick: extra tabulation.

File: Sources/VLCMovieViewController.m [hunk 4]
```diff
@@ -1350,24 +1449,40 @@ - (void)toggleChapterAndTitleSelector
...
+                         animations:^{
+                             self->_multiSelectionView.hidden = YES;
+                         }
+                         completion:^(BOOL finished){
+                         }];
+        [self _resetIdleTimer];
+    #else
+        [UIView animateWithDuration:.2
+                         animations:^{
+                             self->_videoOptionsControlBar.hidden = YES;
```
Comment (Mikanbu) [inline Sources/VLCMovieViewController.m]: I believe we could avoid duplication here and make the check inside the animations block.

File: Sources/VLCMovieViewController.m [hunk 5]
```diff
@@ -1662,12 +1777,20 @@ - (BOOL)shouldAutorotate
 - (void)viewWillTransitionToSize:(CGSize)size withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator
 {
     [super viewWillTransitionToSize:size withTransitionCoordinator:coordinator];
-
+
```
Comment (Mikanbu) [inline Sources/VLCMovieViewController.m]: Cosmetic nit-pick: Extra whitespace.

File: Sources/VLCMovieViewController.m [hunk 6]
```diff
@@ -1662,12 +1777,20 @@ - (BOOL)shouldAutorotate
...
-
+    
     if (UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPhone) {
-
         [coordinator animateAlongsideTransition:^(id<UIViewControllerTransitionCoordinatorContext>  _Nonnull context) {
-                if (self.artworkImageView.image)
-                    self.trackNameLabel.hidden = YES;
+            if(self->_videoOptionsControlBar.hidden == NO) {
+                self->_videoOptionsControlBar.hidden = YES;
+                if(UIInterfaceOrientationIsLandscape([[UIApplication sharedApplication] statusBarOrientation])) {
```
Comment (Mikanbu) [inline Sources/VLCMovieViewController.m]: Cosmetic nit-pick: Missing space after keywords.

File: Sources/VLCMovieViewController.m [hunk 7]
```diff
@@ -1746,4 +1869,24 @@ - (void)playbackSpeedViewSleepTimerHit:(PlaybackSpeedView *)playbackSpeedView
     [self showSleepTimer];
 }
```
Comment (Mikanbu) [inline Sources/VLCMovieViewController.m]: We could perhaps add a `#pragma mark - VLCVideoOptionsControlBarDelegate` here.

File: Sources/VLCVideoOptionsControlBar.swift [hunk 8]
```diff
@@ -0,0 +1,142 @@
+/*****************************************************************************
+ * VLCMediaViewEditCell.swift
+ *
+ * Copyright © 2018 VLC authors and VideoLAN
+ * Copyright © 2018 Videolabs
```
Comment (Mikanbu) [inline Sources/VLCVideoOptionsControlBar.swift]: Header naming should reflect the filename. 😅 
Additionally, bump the year and remove the Videolabs copyright.

File: Sources/VLCVideoOptionsControlBar.swift [hunk 9]
```diff
@@ -0,0 +1,142 @@
...
+ *
+ * Authors: Robert Gordon <robwaynegordon@gmail.com>
+ *
+ *
+ * Refer to the COPYING file of the official project for license.
+ *****************************************************************************/
+
+@objc protocol VLCVideoOptionsControlBarDelegate: class {
+    func didToggleFullScreen(_ optionsBar: VLCVideoOptionsControlBar)
```
Comment (Mikanbu) [inline Sources/VLCVideoOptionsControlBar.swift]: I believe the protocol name in Swift shouldn't have the VLC prefix like so: `VideoOptionsControlBarDelegate`.
Though, when expositing it to Objective-C, we should add the prefix.
Therefore we should have something like this:
```
@objc(VLCVideoOptionsControlBarDelegate)
protocol VideoOptionsControlBarDelegate...
```

Same remark for the filename.

File: Sources/VLCVideoOptionsControlBar.swift [hunk 10]
```diff
@@ -0,0 +1,142 @@
...
+    func didToggleRepeat(_ optionsBar: VLCVideoOptionsControlBar)
+    func didSelectSubtitle(_ optionsBar: VLCVideoOptionsControlBar)
+    func didSelectMoreOptions(_ optionsBar: VLCVideoOptionsControlBar)
+    func didToggleInterfaceLock(_ optionsBar: VLCVideoOptionsControlBar)
+}
+
+class VLCVideoOptionsControlBar: UIStackView {
```
Comment (Mikanbu) [inline Sources/VLCVideoOptionsControlBar.swift]: Same remark as the protocol.

File: Sources/VLCVideoOptionsControlBar.swift [hunk 11]
```diff
@@ -0,0 +1,142 @@
...
+    
+    // MARK: Instance variables
+    @objc var delegate: VLCVideoOptionsControlBarDelegate?
```
Comment (Mikanbu) [inline Sources/VLCVideoOptionsControlBar.swift]: This delegate should be set to weak in order to avoid potential retain cycles.

File: Sources/VLCVideoOptionsControlBar.swift [hunk 12]
```diff
@@ -0,0 +1,142 @@
...
+    private var __repeatMode: VLCRepeatMode = .doNotRepeat
```
Comment (Mikanbu) [inline Sources/VLCVideoOptionsControlBar.swift]: Nit-pick: In VLC-iOS we don't mark private variable with `__`.

File: Sources/VLCVideoOptionsControlBar.swift [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+        subbutton.tintColor = .white
+        // TODO: add accessability options for selectingSubtitleButton
+        return subbutton
+    }()
+    
+    @objc var repeatButton: UIButton = {
+        var rptButton = UIButton(type: .system)
+        rptButton.addTarget(self, action: #selector(toggleRepeat), for: .touchUpInside)
+        rptButton.setImage(UIImage(named: "no-repeat-new"), for: .normal)
+        rptButton.tintColor = .orange
```
Comment (Mikanbu) [inline Sources/VLCVideoOptionsControlBar.swift]: Is there a specific reason of why this is set to orange?

File: VLC.xcodeproj/project.pbxproj [hunk 14]
```diff
@@ -2345,7 +2348,7 @@
 						};
 					};
 					7D94FCDA16DE7D1000F2623B = {
-						DevelopmentTeam = 75GAHG3SZQ;
+						DevelopmentTeam = 79H4ABEPC6;
```
Comment (Mikanbu) [inline VLC.xcodeproj/project.pbxproj]: DevelopmentTeam related changes shouldn't be included in this PR.

File: Sources/VLCVideoOptionsControlBar.swift (hunk 8 above)
Comment (robwayne) [reply Sources/VLCVideoOptionsControlBar.swift]: Haha copied that and forgot to change it 😅😂

File: Sources/VLCVideoOptionsControlBar.swift (hunk 13 above)
Comment (robwayne) [reply Sources/VLCVideoOptionsControlBar.swift]: Nope, my mistake

File: Sources/VLCMovieViewController.m [hunk 15]
```diff
@@ -1233,39 +1292,65 @@ - (IBAction)sleepTimerAction:(id)sender
...
+                         completion:^(BOOL finished){
+                         }];
+        #else
+            if (_videoOptionsControlBar.hidden == NO) {
+                [UIView animateWithDuration:.3
+                                 animations:^{
+                                     self->_videoOptionsControlBar.hidden = YES;
+                                 }
+                                 completion:^(BOOL finished){
+                                 }];
```
Comment (carolanitz) [inline Sources/VLCMovieViewController.m]: completion: nil];

File: Sources/VLCMovieViewController.m [hunk 16]
```diff
@@ -1233,39 +1292,65 @@ - (IBAction)sleepTimerAction:(id)sender
...
+            /* animate */
+           _videoOptionsControlBar.frame = CGRectMake(workFrame.origin.x, workFrame.origin.y + workFrame.size.height, workFrame.size.width, 0.);
+            [UIView animateWithDuration:.3
+                             animations:^{
+                                 self->_videoOptionsControlBar.frame = workFrame;
+                                 self->_videoOptionsControlBar.hidden = NO;
+                             }
+                             completion:^(BOOL finished){
+                             }];
+        #endif
```
Comment (carolanitz) [inline Sources/VLCMovieViewController.m]: this code should get easier with constraints

File: Sources/VLCMovieViewController.m [hunk 17]
```diff
@@ -1665,12 +1762,19 @@ - (BOOL)shouldAutorotate
...
         [coordinator animateAlongsideTransition:^(id<UIViewControllerTransitionCoordinatorContext>  _Nonnull context) {
-                if (self.artworkImageView.image)
-                    self.trackNameLabel.hidden = YES;
+            if (self->_videoOptionsControlBar.hidden == NO) {
+                self->_videoOptionsControlBar.hidden = YES;
+                if (UIInterfaceOrientationIsLandscape([[UIApplication sharedApplication] statusBarOrientation])) {
+                    self->_videoOptionsControlBar.orientationAxis = UILayoutConstraintAxisVertical;
+                } else {
+                    self->_videoOptionsControlBar.orientationAxis = UILayoutConstraintAxisVertical;
+                }
```
Comment (carolanitz) [inline Sources/VLCMovieViewController.m]: both times vertical?

File: Sources/VLCMovieViewController.m [hunk 18]
```diff
@@ -1749,4 +1853,26 @@ - (void)playbackSpeedViewSleepTimerHit:(PlaybackSpeedView *)playbackSpeedView
...
+}
+
+- (void)didToggleInterfaceLock:(VLCVideoOptionsControlBar * _Nonnull)optionsBar {
+    NSLog(@"interFaceLock Selected");
+}
+
+- (void)didToggleRepeat:(VLCVideoOptionsControlBar * _Nonnull)optionsBar {
+    NSLog(@"repeatButtonToggled Selected");
+}
+
```
Comment (carolanitz) [inline Sources/VLCMovieViewController.m]: 👍

File: Sources/VideoOptionsControlBar.swift [hunk 19]
```diff
@@ -0,0 +1,147 @@
...
+            axis = newValue
+            let minX: CGFloat = frame.maxX - frame.size.height
+            let minY: CGFloat = frame.maxY - frame.size.width
+            frame = CGRect(x: minX, y: minY, width: frame.size.height, height:frame.size.width)
+        }
+        
+        get {
+            return axis
+        }
+    }
```
Comment (carolanitz) [inline Sources/VideoOptionsControlBar.swift]: if we change to constraints I think we will not need this part though

File: VLC.xcodeproj/project.pbxproj [hunk 20]
```diff
@@ -3808,6 +3814,7 @@
 				CURRENT_PROJECT_VERSION = 303;
 				DEAD_CODE_STRIPPING = NO;
 				DEBUG_INFORMATION_FORMAT = "dwarf-with-dsym";
+				DEVELOPMENT_TEAM = 79H4ABEPC6;
```
Comment (carolanitz) [inline VLC.xcodeproj/project.pbxproj]: oh also make sure to not have these changes in your PR ^

File: Sources/VideoOptionsControlBar.swift [hunk 21]
```diff
@@ -0,0 +1,147 @@
...
+            switch newValue {
+            case .repeatCurrentItem:
+                repeatButton.setImage(UIImage(named: "repeatOne-new"), for: .normal)
+                break
+            case .repeatAllItems:
+                repeatButton.setImage(UIImage(named: "repeat-new"), for: .normal)
+                break
+            default: // no repeat
+                repeatButton.setImage(UIImage(named: "no-repeat-new"), for: .normal)
+                break
```
[the remark this answers is not in the document]
Comment (robwayne) [reply Sources/VideoOptionsControlBar.swift]: Oh yeah true, I thought I had read somewhere that it's good to have them. I guess I mixed it up

File: Sources/VideoOptionsControlBar.swift [hunk 22]
```diff
@@ -0,0 +1,147 @@
...
+        super.init(coder: aDecoder)
+    }
+    
+    private func setupButtonTargets() {
+        toggleFullScreenButton.addTarget(self, action: #selector(toggleFullscreen), for: .touchUpInside)
+        interfaceLockButton.addTarget(self, action: #selector(toggleInterfaceLock), for: .touchUpInside)
+        repeatButton.addTarget(self, action: #selector(toggleRepeat), for: .touchUpInside)
+        selectSubtitleButton.addTarget(self, action: #selector(selectSubtitle), for: .touchUpInside)
+        moreOptionsButton.addTarget(self, action: #selector(selectMoreOptions), for: .touchUpInside)
+    }
```
[the remark this answers is not in the document]
Comment (robwayne) [reply Sources/VideoOptionsControlBar.swift]: So I did do that initially, however, for some reason there was a bug where the actions were not being added to the buttons because `self` had not been created yet. There was some weird behavior happening that @Mikanbu and I couldn't understand until I did this.

File: Sources/VLCMovieViewController.m (hunk 17 above)
Comment (robwayne) [reply Sources/VLCMovieViewController.m]: 😅

File: Sources/VideoOptionsControlBar.swift [hunk 23]
```diff
@@ -0,0 +1,147 @@
...
+protocol VideoOptionsControlBarDelegate: class {
+    func didToggleFullScreen(_ optionsBar: VideoOptionsControlBar)
+    func didToggleRepeat(_ optionsBar: VideoOptionsControlBar)
+    func didSelectSubtitle(_ optionsBar: VideoOptionsControlBar)
+    func didSelectMoreOptions(_ optionsBar: VideoOptionsControlBar)
+    func didToggleInterfaceLock(_ optionsBar: VideoOptionsControlBar)
+}
+
+@objc (VLCVideoOptionsControlBar)
+class VideoOptionsControlBar: UIStackView {
```
[the remark this answers is not in the document]
Comment (robwayne) [reply Sources/VideoOptionsControlBar.swift]: 🤯

File: Sources/VideoOptionsControlBar.swift (hunk 22 above)
[the remark this answers is not in the document]
Comment (carolanitz) [reply Sources/VideoOptionsControlBar.swift]: probably because they're not lazily instantiated

File: Sources/VideoOptionsControlBar.swift (hunk 19 above)
Comment (robwayne) [reply Sources/VideoOptionsControlBar.swift]: I decided to just leave this in for now because the next phase would be to add the constraints and then I can just replace it with the intended constraints.

File: Sources/VLCMovieViewController.m [hunk 24]
```diff
@@ -471,31 +485,43 @@ - (void)viewDidLayoutSubviews
...
+        CGRect videoOptionsBarFrame = CGRectMake(0., 0., 24., 120.);
+        CGFloat x,y;
+        x = controllerPanelFrame.size.width - videoOptionsBarFrame.size.width;
+        y = controllerPanelFrame.origin.y - videoOptionsBarFrame.size.height;
+        videoOptionsBarFrame = CGRectMake(x,y,videoOptionsBarFrame.size.width,videoOptionsBarFrame.size.height);
+        _videoOptionsControlBar.frame = videoOptionsBarFrame;
+        if (UIInterfaceOrientationIsLandscape([[UIApplication sharedApplication] statusBarOrientation])) {
+            _videoOptionsControlBar.orientationAxis = UILayoutConstraintAxisHorizontal;
+        }
+    #endif
```
[the remark this answers is not in the document]
Comment (robwayne) [reply Sources/VLCMovieViewController.m]: Can I possibly do this in the next PR? I have made the changes and split this PR into three main PRs so it would be more convenient in the next PR to make this change while adding the proper constraints
